crates/scarb-api/src/artifacts.rs

```diff
@@ -12,34 +13,53 @@ mod deserialized;
 mod representation;
 
 /// Contains compiled Starknet artifacts
-#[derive(Debug, PartialEq, Clone)]
+#[derive(Debug, Clone)]
 pub struct StarknetContractArtifacts {
     /// Compiled sierra code
     pub sierra: String,
     /// Compiled casm code
     pub casm: String,
+
+    /// Optional AOT compiled native executor
+    pub executor: Option<AotContractExecutor>,
+}
```

Comment:
To be 100% idiomatic, I think this struct should be moved to forge crate or somewhere else so that that impl is owned by the user. It doesn't make much sense for Scarb api to own a struct for `StarknetContractArtifacts` with details about native compilation if scarb knows nothing about that.

Reply:
And while we are at it, having casm here if compilation is done by USC not scarb is also weird.

Reply:
Unless refactoring this is an extremely big lift, I think now might be a good moment to finally untangle that.